Appleseed install at IE

andrewkaufman opened the pull request:
This gets IECoreAppleseed installing at IE for the first time. I did expose some new SConstruct options, but I hope that's ok because they seem like options we'd want in a future that uses OSL and OIIO anyways.

johnhaddon commented on contrib/IECoreAppleseed/src/IECoreAppleseed/RendererImplementation.cpp:
```diff
@@ -924,10 +924,10 @@ void IECoreAppleseed::RendererImplementation::setCamera( const string &name, Cam
-	crop.min[1] = cropWindow->readable().min.y * ( res[1] - 1 );
-	crop.max[0] = cropWindow->readable().max.x * ( res[0] - 1 );
-	crop.max[1] = cropWindow->readable().max.y * ( res[1] - 1 );
+	crop.min[0] = (int)cropWindow->readable().min.x * ( res[0] - 1 );
```
Have you tested this, or do we have unit test coverage for it? I _think_ the cast to int binds to `cropWindow->readable().min.x` rather than the result of the whole calculation. Meaning it's going to either be 0 or 1 before the multiplication, meaning crop windows would be broken.

andrewkaufman commented:
You caught me, I hadn't actually ran the tests, and you're right it broke the crop. I've fixed that now and rebased (just the last 2 commits are new). I did run the tests, but they don't currently cover the crop, and I spent a little time trying to write a test and failing to figure out how/where to confirm the crop using the Appleseed python module... I did verify in Gaffer that the crop works again now though...

johnhaddon commented:
Looks good to me. If I'm not around once the Travis tests have passed, feel free to merge away...

est77 commented:
> I spent a little time trying to write a test and failing to figure out how/where to confirm the crop using the Appleseed python module...

I just checked and it's not possible to get or set a crop window from python. 
We'll fix it and I'll add a test in IECoreAppleseed.

andrewkaufman commented:
OK, that'd be great, thanks Est. I'll merge this now anyways since I manually checked it

est77 commented:
I added the missing crop window methods here:
https://github.com/appleseedhq/appleseed/pull/1009

dictoon commented:
Just merged Est's PR into appleseed master.

andrewkaufman commented:
Thanks guys. One thing I noticed (nothing todo with the PR really) is that the geo I render in Appleseed is coming out a bit blocky/faceted... is IECoreAppleseed maybe not using the MeshPrimitive.interpolation info to subdivide at render time? If not, fair enough since it doesn't seem to be well documented at all, but "linear" is the default, and "catmullClark" is the interpolation setting to tell the renderer to subdivide.

Actually, I just grepped and I guess I have my answer. Here's how we're doing it in Arnold (https://github.com/ImageEngine/cortex/blob/0615f6e2ad6df45e6abff3d142a26e176d4eb745/contrib/IECoreArnold/src/IECoreArnold/MeshAlgo.cpp#L92-L96) and 3delight (https://github.com/ImageEngine/cortex/blob/c2fff0974e5c0ea0462e3bcd32218f0c5184f702/src/IECoreRI/RendererImplementation.cpp#L1837-L1885) anyways.

est77 commented:
Appleseed does not render subdivision surfaces at the moment. We have plans to support them 
and probably displacement mapping too in a future release.

andrewkaufman commented:
Ok, thanks, no worries. I have a Gaffer node that uses OpenSubdiv, so I can just bake it in for now.
